### DISCUSSION TITLE
Is the time taken by a complex query in sql run using queryRaw and normal raw sql query the same or does Prisma have to run anything under the hood?

### DISCUSSION THREAD

**Discussion by Akhileshpm** [prisma/prisma#24461] [Q&A]
When I started out using I really never thought ORMs were really slow(no offence).  Only when I completed the application setup did I notice the complaints of developers about the performance. However since Prisma provides raw query support, I have started using it for running complex queries. But I still wonder if the difference in time taken to execute these queries by raw SQL and query raw is negligible.

**Comment by ludralph** [comment]
Hi @Akhileshpm 👋 

Thank you for raising this question

The performance difference between raw SQL and `queryRaw` in Prisma is usually negligible for most practical purposes. In terms of execution time, both raw SQL and `queryRaw` send the query to the database, which executes it. The execution time on the database side is identical. Both approaches involve network transmission of queries and results. This overhead is the same.
Prisma's queryRaw introduces only a slight overhead for interfacing with the database driver.  It does introduce some performance overhead compared to raw SQL queries. However, for many applications, the trade-off between convenience and performance might be acceptable.

**Reply by Akhileshpm** [reply]
Thanks, @ludralph for the insights, I was looking exactly for this.

**Reply by janpio** [reply]
Also, make sure that you are not falling back to raw queries without a reason - the SQL queries generated by your more convenient APIs are usually perfectly fine, and often as fast or even faster what one would naively write by hand. Hundreds of thousands of developers are using these APIs, and the performance problem only apply to specific use cases - which after being reported to us, we usually can and do also fix.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
